leif/main.cpp

```diff
@@ -25,9 +27,92 @@
 
 #ifdef QT_DEBUG
 #include "log/consolelogger.h"
+
 //#include "log/logfilterbyfile.h"
 #endif // ! QT_DEBUG
 
+QString macOSXAppBundlePath()
+{
+#ifdef Q_OS_MAC
+    QDir dir = QDir ( QCoreApplication::applicationDirPath() );
+    dir.cdUp();
+    dir.cdUp();
+    QString absolutePath = dir.absolutePath();
+    // absolutePath will contain a "/" at the end,
+    // but we want the clean path to the .app bundle
+    if ( absolutePath.length() > 0 && absolutePath.right(1) == "/" ) {
+        absolutePath.chop(1);
+    }
+    return absolutePath;
+#else
+    return "";
+#endif
+}
+
+
+QString macOSXAppBundleName()
+{
+#ifdef Q_OS_MAC
+    QString bundlePath = macOSXAppBundlePath();
+    QFileInfo fileInfo(bundlePath);
+    return fileInfo.baseName();
+#else
+    return "";
+#endif
+}
+
+bool setAppToStartAutomatically ( bool startAutomatically ) {
+#if defined ( Q_OS_MAC )
+
+    // Remove any existing login entry for this app first, in case there was one
+    // from a previous installation, that may be under a different launch path.
+    {
+        QStringList args;
+        args << "-e tell application \"System Events\" to delete login item\""
+                    + macOSXAppBundleName() + "\"";
+
+        QProcess::execute("osascript", args);
+    }
+
+    // Now install the login item, if needed.
+    if ( startAutomatically )
+    {
+        QStringList args;
+        args << "-e tell application \"System Events\" to make login item at end with properties {path:\"" + macOSXAppBundlePath() + "\", hidden:false}";
+
+        QProcess::execute("osascript", args);
+    }
+
+#elif defined ( Q_OS_WIN )
```

Comment:
The windows part can also be removed.

Reply:
Thanks @scharsig, I've made those changes. Just need to work out the UI part next